### ISSUE TITLE
Inference transform

### ISSUE THREAD

**Issue by pvnieo** [nicolas-chaulet/torch-points3d#416]
Hi,

Thanks for the great library.

I'm following the KPConv notebook, and I'm having a question about the `inference_transform`. 

![image](https://user-images.githubusercontent.com/22404728/91040162-0c638200-e60e-11ea-8ada-dffcc38f1e03.png)

Actually, I want to know how they are constructed (because they are not mentioned in the yaml config), also, for what they are used?

Also, the validation set does not have a transform because it's not used during the training?

Thanks in advance!

**Comment by humanpose1** [comment]
Hi @pvnieo ,

The `inference_transform` is the `pre_transform` and the `test_transform`. It is built automatically (see [here).](https://github.com/nicolas-chaulet/torch-points3d/blob/225ca62320436a2fa1a3ea3503ebe692e001a6ff/torch_points3d/datasets/base_dataset.py#L131) It is useful for demos, when pre_transform has not been applied to data.

As for the the validation set, it's not used  during the training set in that case. It's just an exploration notebook to see how the framework and kpconv work. But in other cases (see the [conf](https://github.com/nicolas-chaulet/torch-points3d/tree/master/conf/data/segmentation) files), the `val_transform` is defined.

Don't hesitate if you have other questions !

**Comment by pvnieo** [comment]
@humanpose1 thank you for your response. It's clear now!

I have another question concerning the implementation of KPConv. In fact, I want to know if there is any randomness of the model when used for test or validation (with `.eval()`)? Actually, I'm testing the same model, with the same data, in the same order, and I'm getting a different result each time, up to 0.2 difference in validation loss, what would be the source of this?

**Comment by humanpose1** [comment]
normally, there isn't any randomness in KPConv for the validation. 
Can you please send us an example code ?

**Comment by nicolas-chaulet** [comment]
Closing for now, let us know if you still ahve troubles!
